packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -20,7 +20,17 @@ import { AssetManifestBuilder } from '../util/asset-manifest-builder';
 import { determineAllowCrossAccountAssetPublishing } from './util/checks';
 import { publishAssets } from '../util/asset-publishing';
 
-export interface DeployStackResult {
+export type DeployStackResult =
+  // Regular result
+  | RegularDeployStackResult
+  // The stack is currently in a failpaused state, and needs to be rolled back before the deployment
+  | { type: 'failpaused-need-rollback-first'; reason: 'not-norollback' | 'replacement' }
+  // The upcoming change has a replacement, which requires deploying without --no-rollback.
+  | { type: 'replacement-requires-norollback' }
+  ;
+
+export interface RegularDeployStackResult {
+  readonly type: 'did-deploy-stack';
```

Comment:
The type system is doing some heavy lifting here. Would a constant prevent having to types this multiple times?

Reply:
> The type system is doing some heavy lifting here

Yep! Nice, isn't it?

> Would a constant prevent having to types this so many times?

It really doesn't make a difference. This is a string-as-a-type, and therefore just as safe to use as `DID_DEPLOY_STACK`. The only difference would be 2 keystrokes.

Reply:
Cool! Does it auto-complete as well? 

I'm all for avoiding enums to be honest.